rust/automerge/src/read.rs

```diff
@@ -196,4 +196,7 @@ pub trait ReadDoc {
 
     /// Get a change by its hash.
     fn get_change_by_hash(&self, hash: &ChangeHash) -> Option<&Change>;
+
+    /// Get the hash of the change that contains the given opid.
+    fn hash_for_opid(&self, opid: &ExId) -> Option<ChangeHash>;
```

Comment:
I'd like to NOT add new functions to ReadDoc if it can be avoided.  In fact I'd like to remove `get_change_by_hash` and `get_missing_deps` from the trait.   Having it be a public function on `Automerge` and `Autocommit` should be sufficient.